src/components/buttons/HdButton.vue

```diff
@@ -47,20 +57,45 @@ export default {
       default: '',
     },
   },
+  data() {
+    return {
+      contentObserver: null,
+      isIconButton: false,
+    };
+  },
   computed: {
     computedClasses() {
       const baseClass = 'btn';
-      const classes = [baseClass];
-
-      if (this.modifier) {
-        classes.push(`${baseClass}--${this.modifier}`);
-      }
-
-      if (this.isInDarkBackground) {
-        classes.push(`${baseClass}--dark-background`);
-      }
+      return [
+        baseClass,
+        {
+          [`${baseClass}--${this.modifier}`]: this.modifier,
+          [`${baseClass}--dark-background`]: this.isInDarkBackground,
+          [`${baseClass}--icon-button`]: this.isIconButton,
+        },
+      ];
+    },
+  },
+  mounted() {
+    // We set an observer to watch the content of the button
+    // to know when to switch to the icon button mode
+    this.contentObserver = new MutationObserver(() => {
```

Comment:
I understood that you are doing this to automatically detect if the button has some content to display.

But I have some questions regarding this:
- Have you tried something like `this.$slots.default`? Because it returns `false` if no slot was provided, so you could replace the MutationObserver
  - Or did you go with this approach to be more strict?
- Is not possible to just use CSS to solve this problem, using something like the [`:empty` selector](https://developer.mozilla.org/en-US/docs/Web/CSS/:empty)

Reply:
Thank you for the review @kuroski!
- Yes, I had the `this.$slots.default` in mind, and you can see me using it in the first attempt [here](https://github.com/homeday-de/homeday-blocks/pull/552/commits/a84ed91d949e8b8043b8e6d10be1891260c08893#diff-3101f32ed834ae53815e7bf66063cc3aR52). The problem I faced with it is that $slots is not reactive, so if the content changes inside after the component is mounted, there is no way to detect that, and this is important because I guess some button will change content based on the viewport. I'm observing only the content part of the button, so that shouldn't cause any performance issues.
- The problem with `:empty` is that it doesn't handle whitespace properly, and won't solve the issue with nested elements.

Reply:
Hmm, I see...
Thanks a lot for the explanation =DDD